src/PhysicsEngine.cpp

```diff
@@ -185,3 +184,257 @@ void ForwardEuler::march_step(double tStamp, double tStep) {
+void RungeKutta::march_step(double tStamp, double tStep) {
+
+    /*************** Retrieve Instantaneous Rocket Parameters *****************/
+
+    Vector3 pos_if = rocket_.get_r_vect();
+    Vector3 vel_if = rocket_.get_r_dot();
+    Vector3 accel_if = rocket_.get_r_ddot();
+    Vector3 ang_vel_if = rocket_.get_w_vect();
+    Vector3 ang_accel_if = rocket_.get_w_dot();
+    
+    Quaternion<double> orient = rocket_.get_q_ornt();
+
+    double inertia[9];         // moments of inertia
+    rocket_.get_I(inertia);
+
+    double mass = rocket_.get_mass();
+
+    /********************* Calculate Intermediate States ***********************/
+    // Each state is used to calculate the next state
+
+    State k1 {vel_if, accel_if, ang_vel_if, ang_accel_if};           // Reformating of the initial state of the rocket
+    State k2 = calc_state(tStamp, 0.5 * tStep, k1);                  // Calculated with Euler step using half tStep and initial state
+    State k3 = calc_state(tStamp + (0.5 * tStep), 0.5 * tStep, k2);  // Calculated with Euler step using half tStep and k2 state
+    State k4 = calc_state(tStamp + (0.5 * tStep), tStep, k3);        // Calculated with Euler step using full tStep and k3 state
+
+    /********************** Perform Runge-Kutta Method ************************/
+
+    // calculate weighted averages
+    Vector3 vel_avg = (k1.vel + (2 * k2.vel) + (2 * k3.vel) + k4.vel) / 6;
+    Vector3 accel_avg = (k1.accel + (2 * k2.accel) + (2 * k3.accel) + k4.accel) / 6;
+    Vector3 ang_vel_avg = (k1.ang_vel + (2 * k2.ang_vel) + (2 * k3.ang_vel) + k4.ang_vel) / 6;
+    Vector3 ang_accel_avg = (k1.ang_accel + (2 * k2.ang_accel) + (2 * k3.ang_accel) + k4.ang_accel) / 6;
+
+    // calculate rocket data based on average values instaed of initial
+    Vector3 net_force_if = calc_net_force(tStamp, vel_avg);
+    Vector3 net_torque_if = calc_net_torque(vel_avg, ang_vel_avg);
+
+    pos_if += tStep * vel_avg;
+    vel_if += tStep * accel_avg;
+    accel_if = net_force_if / mass;
+
+    ang_vel_if += tStep * ang_accel_avg;
+    ang_accel_if.x = net_torque_if.x / inertia[0];
+    ang_accel_if.y = net_torque_if.y / inertia[4];
+    ang_accel_if.z = net_torque_if.z / inertia[8];
+
+    orient = calc_orient(tStep, ang_vel_avg, orient);
+
+    //---- Lauch Rail ----
+    // very basic implementation
+    if (pos_if.magnitude() < 4.50) {
+        ang_vel_if.x = 0;
+        ang_vel_if.y = 0;
+        ang_vel_if.z = 0;
+        ang_accel_if.x = 0;
+        ang_accel_if.y = 0;
+        ang_accel_if.z = 0;
+    }
+
+    //---- Set Values ----
+    rocket_.set_r_vect(pos_if);
+    rocket_.set_r_dot(vel_if);
+    rocket_.set_r_ddot(accel_if);
+    rocket_.set_w_vect(ang_vel_if);
+    rocket_.set_w_dot(ang_accel_if);
+    rocket_.set_f_net(net_force_if);
+    rocket_.set_t_net(net_torque_if);
+    rocket_.set_q_ornt(orient);
+
+}
+
+/**
+ * @brief Takes in time and velocity to calculate net force
+ * 
+ * @param tStamp 
+ * @param vel_if 
+ * @return Vector3
```

Comment:
We can see what the parameters and returns are ourselves lol. Please add a short description of what these are